### PR TITLE
scroll performance engineering blog article

### PR DESCRIPTION
**Pull request by uBhappy2**

### CODE REVIEW

File: _drafts/2015-12-16-improving-scrolling-performance-of-a-UICollectionView.markdown [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
+layout:       post
+title:        Additional tips on improving scrolling performance of a UICollectionView
+author:       amit_rao
+summary:      Recently we improved the scrolling performance of the WeWork iOS app with the goal of getting as close to 60 fps consistently as possible. The blog post describes my experience solving the issue and adding to some of the existing online resources related to the issue.
+image:		  http://res.cloudinary.com/wework/image/upload/v1430060208/engineering/recruiting-through-gamification.jpg      
+categories:   engineering
+---
+
+#### What does optimizing scrolling performance mean and why is it important?
```
Comment (i8ramin) [inline _drafts/2015-12-16-improving-scrolling-performance-of-a-UICollectionView.markdown]: Not sure if you need a header here. the paragraph below is very short and can probably stand on its own as an intro summary of the article. You can even repeat some of what you have in the `summary` section above

File: _drafts/2015-12-16-improving-scrolling-performance-of-a-UICollectionView.markdown [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
...aded  without using `imageNamed` or the resolution of the image is too high e.g. a large image being loaded into a thumbnail view.  What is interesting in the video is that the Apple engineer recommends using the platform's built in caching support, async drawing support & flattening support before pulling out the heavy tools. It is also important to make sure one is reusing cells etc. In this case eliminating blending of the layers in the various subviews of the `UICollectionViewCell` gave the most bang for the buck and was a solution we could live with. The general methodology is to determine if it a CPU or GPU bound issue using the CA Instrument. Come up with a theory and baseline, then make code changes and measure again. See if there is improvement and record all measurements. The WWDC video recommends some other tips worth checking out and  speculative caching is another powerful technique. You might want to check out `NSCache/NSPurgeable` especially for images. Facebook’s *AsyncDisplayKit* might be worth checking out especially if one is developing an app from scratch (We did not use it). Here are the frame rates after blending all the CA layers in the collection view cells.
```
Comment (i8ramin) [inline _drafts/2015-12-16-improving-scrolling-performance-of-a-UICollectionView.markdown]: this is a very long paragraph. you should probably break it up...

File: _drafts/2015-12-16-improving-scrolling-performance-of-a-UICollectionView.markdown [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+To improve the thumbnail insertion if you have significant number of thumbnail images, consider using `NSCache/NSPurgeable` for the thumbnail images. In other words, create the thumbnail image on the fly and store the result in a cache (speculative caching) based on a key using the cell or record ID. Then the next time the image is needed, rather than recalculating/resizing you re-use the cached image. If it has been purged, you recalculate.  This is pretty straightforward and can be used for all scrolling which has a number of images which are expensive to either generate or retrieve. It is also what the Apple home screen uses for all the App Icons.
+
+There is a WWDC talk Re: [how Apple implemented the home screen scrolling](https://developer.apple.com/videos/play/wwdc2015-212/) and all of the parameters they had to consider such as cache size, memory usage vs cache usage. It is an excellent talk for understanding `NSCache/NSPurgeable`.
+
+Hope this helps.
```
Comment (i8ramin) [inline _drafts/2015-12-16-improving-scrolling-performance-of-a-UICollectionView.markdown]: This last part doesn't really add much to the article, unless you wanna elaborate on it and offer help by asking the reader to leave a comment, etc.

Comment (uBhappy2) [inline _drafts/2015-12-16-improving-scrolling-performance-of-a-UICollectionView.markdown]: thanks Ramin for the feedback. I have incorporated your comments. Did you mean all of "Additional Tips" or the last section "There is a WWDC talk ..."?

Comment (i8ramin) [inline _drafts/2015-12-16-improving-scrolling-performance-of-a-UICollectionView.markdown]: just that last line "Hope this helps."

Comment (uBhappy2) [inline _drafts/2015-12-16-improving-scrolling-performance-of-a-UICollectionView.markdown]: Thanks. Done. Is it ok to merge?

**Comment by i8ramin** [comment]
:+1:

**Pull request merged**